pyglet/shapes.py

```diff
@@ -555,15 +561,109 @@ def _get_vertices(self):
+                        angle to the end of the segment.  If None is supplied then the end of the line is
+                        90 degrees to the segment p1->p2.
+                """
+                # These are declared above this function to allow easy reuse from previous runs
+                nonlocal prev_miter
+                nonlocal prev_scale
+                v_np1p2 = Vec2(p2[0] - p1[0], p2[1] - p1[1]).normalize()
+                v_normal = Vec2(-v_np1p2.y,v_np1p2.x)
+
+                # Prep the miter vectors to the normal vector in case it is only one segment
+                v_miter2 = v_normal
+                scale1 = scale2 = self._width / 2.0
+
+                # miter1 is either already computed or the normal
+                v_miter1 = v_normal
+                if prev_miter and prev_scale:
+                    v_miter1 = prev_miter
+                    scale1 = prev_scale
+                elif p0:
+                    # Compute the miter joint vector for the start of the segment
+                    v_np0p1 = Vec2(p1[0] - p0[0], p1[1] - p0[1]).normalize()
+                    v_normal_p0p1 = Vec2(-v_np0p1.y,v_np0p1.x)
+                    # Add the 2 normal vectors and normalize to get miter vector
+                    v_miter1 = Vec2(v_normal_p0p1.x + v_normal.x, v_normal_p0p1.y + v_normal.y).normalize()
+                    scale1 = scale1 / math.sin(math.acos(v_np1p2.dot(v_miter1)))
+
+                if p3:
+                    # Compute the miter joint vector for the end of the segment
+                    v_np2p3 = Vec2(p3[0] - p2[0], p3[1] - p2[1]).normalize()
+                    v_normal_p2p3 = Vec2(-v_np2p3.y,v_np2p3.x)
+                    # Add the 2 normal vectors and normalize to get miter vector
+                    v_miter2 = Vec2(v_normal_p2p3.x + v_normal.x, v_normal_p2p3.y + v_normal.y).normalize()
+                    scale2 = scale2 / math.sin(math.acos(v_np2p3.dot(v_miter2)))
+
+                # Make these tuples instead of Vec2 because accessing
+                # members of Vec2 is suprisingly slow
+                miter1ScaledP = (v_miter1.x * scale1, v_miter1.y * scale1)
+                miter2ScaledP = (v_miter2.x * scale2, v_miter2.y * scale2)
+
+                v1 = (p1[0] + miter1ScaledP[0], p1[1] + miter1ScaledP[1])
+                v2 = (p2[0] + miter2ScaledP[0], p2[1] + miter2ScaledP[1])
+                v3 = (p1[0] - miter1ScaledP[0], p1[1] - miter1ScaledP[1])
+                v4 = (p2[0] + miter2ScaledP[0], p2[1] + miter2ScaledP[1])
+                v5 = (p2[0] - miter2ScaledP[0], p2[1] - miter2ScaledP[1])
+                v6 = (p1[0] - miter1ScaledP[0], p1[1] - miter1ScaledP[1])
+
+                # Setup for next run of function
+                prev_miter = v_miter2
+                prev_scale = scale2
+
+                return (v1[0],v1[1],v2[0],v2[1],v3[0],v3[1],v4[0],v4[1],v5[0],v5[1],v6[0],v6[1])
```

Comment:
Could you add spaces between the items to keep the linters happy?

Reply:
Done